elf/Makefile

```diff
@@ -2,7 +2,7 @@ all: prog.uelf.o
 
 include ../Makefile.common
 
-CFLAGS   = -std=gnu11 -O0 -Wall -Werror -fno-builtin -ffreestanding
+CFLAGS   = -std=gnu11 -Og -Wall -Werror -fno-builtin -ffreestanding
```

Comment:
In my experience `-Og` does interfere with debugging, despite claiming not doing so. It is not as bad as `-O3`, obivously, but can be annoying. This was few GCC versions ago, though, maybe things have improved since then.